mobile/src/main/java/org/openhab/habdroid/core/OpenHabApplication.kt

```diff
@@ -83,6 +86,46 @@ class OpenHabApplication : MultiDexApplication() {
             registerReceiver(listener, IntentFilter(ConnectivityManager.ACTION_RESTRICT_BACKGROUND_CHANGED))
             isSystemDataSaverActive = listener.isSystemDataSaverEnabled(this)
         }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            registerDataAccessAudit()
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    private fun registerDataAccessAudit() {
+        val appOpsCallback = object : AppOpsManager.OnOpNotedCallback() {
+            private fun logPrivateDataAccess(opCode: String, attributionTag: String?, trace: String) {
+                Log.i("DA_$attributionTag", "Operation: $opCode\nStacktrace: $trace")
```

Comment:
Looking at https://developer.android.com/preview/privacy/data-access-auditing I don't think we need to use `attributionTag` at all, as we never call `createAttributionContext`.

Also, is implementing that audit callback really useful for us? I don't think we conciously access private data ourselves (or do we?), so is the goal finding unwated accesses by the libraries we're using?

Reply:
I didn't see a single data access, but IMO it's nice to have and see unwanted activities by libraries. I changed it to log level `e`.